addons/l10n_de/models/datev.py

```diff
@@ -33,3 +33,25 @@ def _post(self, soft=True):
                             raise UserError(_('Account %s does not authorize to have tax %s specified on the line. \
                                 Change the tax used in this invoice or remove all taxes from the account') % (account_name, tax.name))
         return super()._post(soft)
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    def _get_product_accounts(self):
+        """ As taxes with a different rate need a different income/expense account, we add this logic in case people only use
+         invoicing to not be blocked by the above constraint"""
+        result = super(ProductTemplate, self)._get_product_accounts()
+        if self.env.company.country_id.code == "DE" and self.taxes_id.filtered(lambda t: t.company_id == self.env.company):
+            if not self.property_account_income_id:
+                # Search for a corresponding account
+                income_account = self.env['account.account'].search([('internal_group', '=', 'income'), ('deprecated', '=', False),
+                                                               ('tax_ids', 'in', self.taxes_id.filtered(lambda t: t.company_id == self.env.company).ids)], limit=1)
```

Comment:
You should call filtered only once before entering the condition and store it into a variable.

So, I'd reshape the whole condition like this

```suggestion
if self.env.company.country_id.code == "DE" and not self.property_account_income_id:
    company_taxes = self.taxes_id.filtered(lambda t: t.company_id == self.env.company)
    if company_taxes:
        ...
```